src/engine/source/builder/builders/opBuilderHelperNetInfoAddress.cpp

```diff
@@ -0,0 +1,230 @@
+        case Name::BROADCAST: return "/broadcast";
+        default: return "";
+    }
+}
+
+// when isIPv6 == true the third field will be set to 0, otherwise 1 for IPv4
+// agent <agent_id> netaddr save <scan_ID>|<name>|isIPv6=false|add[i]|netm[i]|broad[i]
+// agent 0001 netaddr save 1234|name|0|add0|netm0|broad0
+bool sysNetAddresTableFill(base::Event event,
+                           const std::string& agentId,
+                           const std::string& scan_id,
+                           const std::string& name,
+                           const std::string& ipObjectPath,
+                           std::shared_ptr<wazuhdb::WazuhDB> wdb,
+                           bool isIPv6)
+{
+    // Cheking if AddresArray exists
+    auto address_ar = event->getArray(ipObjectPath + getPath(Name::ADDRESS).data());
+
+    if (!address_ar.has_value())
+    {
+        return false;
+    }
+
+    std::string middleFieldName = isIPv6 ? "IPv6" : "IPv4";
+
+    auto netmaskI = event->getArray(ipObjectPath + getPath(Name::NETMASK).data())
+                        .value_or(std::vector<json::Json>());
+
+    auto broadcastI = event->getArray(ipObjectPath + getPath(Name::BROADCAST).data())
+                          .value_or(std::vector<json::Json>());
+
+    for (size_t i = 0; i != address_ar.value().size(); ++i)
+    {
+        std::optional<std::string> addressIValue {};
+        std::optional<std::string> netmaskIValue {"NULL"};
+        std::optional<std::string> broadcastIValue {"NULL"};
+
+        try
+        {
+            addressIValue = address_ar.value().at(i).getString();
+        }
+        catch (const std::out_of_range& e)
+        {
+            return false;
+        }
+
+        try
+        {
+            netmaskIValue = netmaskI.at(i).getString();
+        }
+        catch (const std::out_of_range& e)
+        {
+            netmaskIValue = "NULL";
+        }
+
+        try
+        {
+            broadcastIValue = broadcastI.at(i).getString().value_or("NULL");
+        }
+        catch (const std::out_of_range& e)
+        {
+            broadcastIValue = "NULL";
+        }
+
+        // We should still check if it's empty because value_or only checks nullopt
+        std::string msg = fmt::format("agent {} netaddr save {}|{}|{}|{}|{}|{}",
+                                      agentId,
+                                      scan_id,
+                                      name,
+                                      isIPv6 ? "1" : "0",
+                                      addressIValue.value(),
+                                      netmaskIValue.value(),
+                                      broadcastIValue.value());
+
+        auto findEventResponse = wdb->tryQueryAndParseResult(msg);
+        if (std::get<0>(findEventResponse) != wazuhdb::QueryResultCodes::OK)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+base::Expression opBuilderHelperNetInfoAddress(const std::any& definition, bool isIPv6)
```

Comment:
Please, if possible, try to join all the returns into a single one.

Reply:
Same as above